tests/h2-support/src/trace.rs

```diff
@@ -1,41 +1,18 @@
-use std::{io, str};
 pub use tracing;
 pub use tracing_subscriber;
 
-pub fn init() -> tracing::dispatcher::DefaultGuard {
-    tracing::subscriber::set_default(
-        tracing_subscriber::fmt()
-            .with_max_level(tracing::Level::TRACE)
-            .with_span_events(tracing_subscriber::fmt::format::FmtSpan::CLOSE)
-            .with_writer(PrintlnWriter { _p: () })
-            .finish(),
-    )
-}
-
-struct PrintlnWriter {
-    _p: (),
-}
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
 
-impl tracing_subscriber::fmt::MakeWriter for PrintlnWriter {
-    type Writer = PrintlnWriter;
-    fn make_writer(&self) -> Self::Writer {
-        PrintlnWriter { _p: () }
-    }
-}
-
-impl io::Write for PrintlnWriter {
-    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
-        let s = str::from_utf8(buf).map_err(|e| io::Error::new(io::ErrorKind::InvalidInput, e))?;
-        println!("{}", s);
-        Ok(s.len())
-    }
-
-    fn write_fmt(&mut self, fmt: std::fmt::Arguments<'_>) -> io::Result<()> {
-        println!("{}", fmt);
-        Ok(())
-    }
+pub fn init() -> tracing::dispatcher::DefaultGuard {
+    let use_colors = atty::is(atty::Stream::Stdout);
+    let layer = tracing_tree::HierarchicalLayer::default()
+        .with_writer(tracing_subscriber::fmt::writer::TestWriter::default())
+        .with_indent_lines(true)
+        .with_ansi(use_colors)
+        .with_targets(true)
+        .with_indent_amount(2)
+        .with_thread_names(true);
```

Comment:
since --- as we discussed --- the thread names are actually not that useful, i'm fine with removing them to make the logs a little less verbose. up to you!
```suggestion
        .with_indent_amount(2);
```